src/main/java/seedu/sellsavvy/logic/parser/AddOrderCommandParser.java

```diff
@@ -0,0 +1,66 @@
+package seedu.sellsavvy.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_COUNT;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_ITEM;
+
+import java.util.stream.Stream;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.logic.commands.ordercommands.AddOrderCommand;
+import seedu.sellsavvy.logic.parser.exceptions.ParseException;
+import seedu.sellsavvy.model.order.Count;
+import seedu.sellsavvy.model.order.Date;
+import seedu.sellsavvy.model.order.Item;
+import seedu.sellsavvy.model.order.Order;
+
+/**
+ * Parses input arguments and creates a new AddOrderCommand object.
+ */
+public class AddOrderCommandParser implements Parser<AddOrderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddOrderCommand
+     * and returns an AddOrderCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AddOrderCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ITEM, PREFIX_DATE, PREFIX_COUNT);
+
+        Index index;
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_ITEM, PREFIX_DATE, PREFIX_COUNT)
```

Comment:
Count should be an optional value right? By default it is count = 1, so in this case PREFIX_COUNT can be omitted for present check.

Reply:
will put in separate issue